sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/CountingSource.java

```diff
@@ -327,8 +372,14 @@ public boolean advance() throws IOException {
       if (Long.MAX_VALUE - source.stride < current) {
         return false;
       }
+      Instant nextTimeToProduce = lastTimeProduced.plus(source.period);
+      if (Instant.now().isBefore(nextTimeToProduce)) {
+        return false;
```

Comment:
That's the correct behavior - if we're not at or beyond the next time we should produce, we have no input available, but may in the future.